java/query/src/main/java/sleeper/query/lambda/SqsQueryRegister.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.query.lambda;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.sqs.AmazonSQS;
+import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.configuration.jars.ObjectFactory;
+import sleeper.configuration.jars.ObjectFactoryException;
+import sleeper.configuration.properties.instance.InstanceProperties;
+import sleeper.configuration.properties.instance.UserDefinedInstanceProperty;
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.configuration.properties.table.TablePropertiesProvider;
+import sleeper.core.iterator.CloseableIterator;
+import sleeper.core.record.Record;
+import sleeper.core.statestore.StateStore;
+import sleeper.core.statestore.StateStoreException;
+import sleeper.io.parquet.utils.HadoopConfigurationProvider;
+import sleeper.query.QueryException;
+import sleeper.query.executor.QueryExecutor;
+import sleeper.query.model.LeafPartitionQuery;
+import sleeper.query.model.Query;
+import sleeper.query.model.QueryOrLeafPartitionQuery;
+import sleeper.query.model.QuerySerDe;
+import sleeper.query.model.output.ResultsOutputInfo;
+import sleeper.query.tracker.DynamoDBQueryTracker;
+import sleeper.query.tracker.QueryStatusReportListeners;
+import sleeper.statestore.StateStoreProvider;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.LEAF_PARTITION_QUERY_QUEUE_URL;
+import static sleeper.configuration.properties.instance.QueryProperty.QUERY_PROCESSOR_LAMBDA_RECORD_RETRIEVAL_THREADS;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+
+public class SqsQueryRegister {
```

Comment:
I find the class name `SqsQueryRegister` a bit confusing. Maybe we could have this called `SqsQueryProcessor`, which would delegate to `SqsLeafPartitionQueryProcessor`?

Reply:
This has been reverted back to the old name